Dnahovskyi/training wiki (#2289)
datagrok-ai/public: help/deploy/complete-setup/configure-auth.md

sidebar_position: 1
---

Datagrok supports such authentication methods:

* [Login-password](#login-password-authentication)
* [Oauth](#oauth-authentication)
* [OpenID](#openid-authentication)
* [LDAP](#ldap-authentication)

To accommodate any enterprise needs, we can develop a customer-specific integration.

> You can enable all methods separately or combined.

Author: Should it be a separate paragraph? Maybe we can integrate this sentence to the first one?

> [Learn more about authentification](../govern/authentication.md).

Author: @ksumag did you consider what I suggested?

> ## Common options

Author: @ksumag did you consider what I suggested?

> ## Common options

Author: Just a thought. Should we even mention these configuration options here? They seem pretty straight-forward from UI already.

> ## Common options

Author: @ksumag, what do you think about removing it at all?
I do not think it is that important to mention it in the beginning of the document

To display something on the login window, set the 'Prompt' option
![Login prompt](login-prompt.png)

Enable the 'Send Welcome Email' option to send welcome emails to new signed-up users.

## Login-password authentication

> This is a basic method to authenticate users with Datagrok.

Author: It is not a good practice to start a paragraph with 'It' or 'This.' And the new sentence is similar to the old one.
Also, It is called General in settings, so it is good to mention it.

To configure login-password authentication for Datagrok:

> 

Author: You should rewrite the instructions from a purpose-location-action perspective, and also use bold for UI elements like Sidebar, to make it easier for users to understand what to do.

The text also contains a lot of 'Datagrok', but it is not worth mentioning. Obviously, it is Datagrok settings, cause you are configuring auth in Datagrok.

Author: The 'email service' should not be a separate paragraph, it is important to mention and set a link to how-to, but it is not worth more than 1 sentence.

Author: Also, you can make the document more personable and easier to understand using 'You can' instead of 'Datagrok can integrate' and so on

Author: You should rewrite the instructions from a purpose-location-action perspective.
For example, instead of 
`Enable **Internal authentication** to use the login-password method` 
write 
`To use the login-password method, in the General section enable 'Internal authentication'`

Author: Same here, parameters are not UI elements, so using bold here is not a good idea. Italic would be better.

Author: The text also contains a lot of 'Datagrok,' which is redundant. And use bold for UI elements.
For example, instead of
`Go to the Datagrok Settings section **Users and Sessions**; this section contains all authentication settings.`
write
`On the left **Sidebar** go to **Settings** > **Users and Sessions**. Authentication settings page appears.`

sidebar_position: 1
---

Datagrok supports authentication using:

* [Login-password](#login-password-authentication)
* [Oauth](#oauth-authentication)
* [OpenID](#openid-authentication)
* [LDAP](#ldap-authentication)

You can enable all authentication methods separately or combined. To accommodate any enterprise needs, we can develop a customer-specific integration. 

[Learn more about authentification](../govern/authentication.md).

## Common options

To display something on the login window, set the **Prompt** option
<p align="center">
  <img src="login-prompt.png" alt="Login prompt" width="300"></img>

> </p>

Author: Docusaurus can not load the image
`<img alt="login-prompt" src={require('./login-prompt.png').default} width="300px"/>`
This way it will work

> Enable the **Send Welcome Email** option to send welcome emails to new signed-up users.

Author: 'Send Welcome Email' is not a UI element, so using bold here is not a good idea. Italic would be better.

## Login-password authentication

To configure login-password authentication for Datagrok:

1. Go to the Datagrok Settings section **Users and Sessions**; this section contains all authentication settings.
2. Enable **Internal authentication** to use the login-password method
3. Signup option is alterable; you can disable it using the **Signup Allowed** option
4. To restrict from which domains people can sign up to the platform, use the **Signup Domains Whitelist** option. You can
   set several domains separated with commas
5. Enable the **Require Email Confirm** option to force people to use working emails.

> To use login-password authentication, it is important to configure an [Email service](configure-smtp.md) that will deliver signup, welcome,

Author: Are you sure 'Email service' should be capitalized?

Author: Let's make it remove some options; I think It can be something like
`that delivers signup and password emails.`

confirmation and forgot password emails.

## Oauth authentication

> Datagrok supports **Google**, **Facebook** and **GitHub** OAUTH authentication.

Author: It is [OAuth](https://oauth.net/), not OAUTH

Author: I would definitely remove bold from here, and mention Google, Facebook and GitHub above in the first paragraph 
https://github.com/datagrok-ai/public/blob/1fd2bde1d5f60e290bd10914197ef5b335fd163a/help/deploy/complete-setup/configure-auth.md?plain=1#L6-L11

> 1. Go to the Datagrok Settings section **Users and Sessions**; this section contains all authentication settings.

Author: We now have different options and only Google Client Id for Google. Can you update the documentation according to the current options, please?

Author: And for all other providers, the settings were also updated. Can you update the doc for them also, please

sidebar_position: 1
---

Datagrok supports authentication using:

* [Login-password](#login-password-authentication)
* [Oauth](#oauth-authentication): 
  * [Google](#google-authentication)
  * [Facebook](#facebook-authentication)
  * [GitHub](#github-authentication)
* [OpenID](#openid-authentication)
* [LDAP](#ldap-authentication)

> You can enable all authentication methods separately or combined.

Author: ```suggestion
You can enable all methods separately or combined.
```

> If supported authentication methods do not work for you, contact us on <info@datagrok.ai> or [our community](https://community.datagrok.ai/) and we will discuss options for your specific case.

Author: ```suggestion
If you need coverage for a specific case, contact us on [community](https://community.datagrok.ai,/), and we will discuss options.
```

> [Learn more about authentification](../../govern/authentication.md).

Author: You can integrate it to the first line.

Author: `Datagrok supports [authentication](../../govern/authentication.md) using:`

## Common options

> You can display a text on the login window, like greetings words etc.

Author: Purpose-location-action.

```suggestion
To display text on the login window, like greetings words, in **Settings** > **Users and Sessions** > General set *Prompt* option.
```

Author: But I still think it is an extra and should be removed from docs.

Reviewer: Okay, so much fight for normal image size...

To do so set the *Prompt* option in **Settings** > **Users and Sessions** > **General**:
<p align="center">
  <img alt="login-prompt" src={require('./login-prompt.png').default} width="300px"/>

> </p>

Author: It does not work, remove align

## Login-password authentication

To configure login-password authentication for Datagrok:

1. On the left **Sidebar** go to **Settings** > **Users and Sessions**. Authentication settings page appears.
2. Enable *Internal authentication* to use the login-password method.

> 3. Adjust the *Signup* option. You can disable or enable it using the *Signup Allowed* and *Allow Oauth Signup* checkboxes.

Author: Purpose-location-action.
```suggestion
3. To disable signup, uncheck the *Signup Allowed* option. To disable OAuth signup, use the *Allow Oauth Signup* option.
```

> 4. For better security you can restrict from which domains people can sign up to the platform. To do so use the *Signup Domains Whitelist* option. You can

Author: 'To do so' is a clear sign that the sentence needs to be rewritten.
For example,
```suggestion
4. To restrict domains people can sign up from, use the *Signup Domains Whitelist* option. Separate several domains with commas.
```

   set several domains separated with commas

> 5. Enable the *Send Welcome Email* option to send welcome emails to new signed-up users.

Author: Purpose-location-action.

> 6. Enable the *Require Email Confirm* option to force users to use working emails.

Author: Purpose-location-action.

Reviewer: @dnillovna , I'll do that, but it looks like non-natural

To use login-password authentication, it is important to configure an [email service](configure-smtp.md) that delivers signup and password emails.

## OAuth authentication

Datagrok supports [Google](#google-authentication), [Facebook](#facebook-authentication) and [GitHub](#github-authentication) OAuth authentication.

### Google authentication

> 1. On the left **Sidebar** go to **Settings** > **Users and Sessions**. Authentication settings page appears

Author: I think we can mention it one on the top of the page, instead of writing it every time

> 2. Enable *Google authentication* to use the Google Oauth method

Author: Purpose-location-action.

> 3. Set *Google Client Id*. You can get it from [Google](https://developers.google.com/identity/oauth2/web/guides/get-google-api-clientid)

Author: Too many 'you can', in this particular case it is redundant
 ```suggestion
3. Set *Google Client Id* obtained from [Google](https://developers.google.com/identity/oauth2/web/guides/get-google-api-clientid)
```

> 4. Make sure the correct **Web Root** is set in the left **Sidebar** > **Settings** > **Admin** section

Author: Purpose-location-action.

> ### Facebook authentication

Author: Same as for Google provider

1. On the left **Sidebar** go to **Settings** > **Users and Sessions**. Authentication settings page appears
2. Enable *Facebook authentication* to use the Facebook Oauth method

> 3. Set *Facebook Client Id* and *Facebook Secret* if applicable. You can get it from [Facebook](https://help.vtex.com/tutorial/adding-a-client-id-and-a-client-secret-to-log-in-with-facebook--3R7rzXWG1GswWOIkYyy8SO)

Author: > if applicable

It is applicable in all cases, so this should be removed

4. Make sure the correct **Web Root** is set in the left **Sidebar** > **Settings** > **Admin** section

> ### Github authentication

Author: Same as for Google provider

1. On the left **Sidebar** go to **Settings** > **Users and Sessions**. Authentication settings page appears

> 2. Enable *GitHub authentication* to use the Facebook Oauth method

Author: You have a copy-paste here, change Facebook to GitHub

3. Set *GitHub Client Id* and *GitHub Secret* if applicable. You can get it from [GiHub](https://episyche.com/blog/how-to-create-oauth-client-id-and-client-secret-for-github)
4. Make sure the correct **Web Root** is set in the left **Sidebar** > **Settings** > **Admin** section

## OpenID authentication

Datagrok supports the OpenID protocol to allow users to be authenticated using OpenID providers, for example, Azure AD.

To setup OpenID authentication:

1. On the left **Sidebar** go to **Settings** > **Users and Sessions**. Authentication settings page appears.
2. Enable *Open Id authentication* to use the OpenID method
3. Get a well-known-configuration route and set it to **Open Id Config Endpoint**. It should look
   like `https://login.datagrok.ai/.well-known/openid-configuration`
4. Set *Open Id Client Id* and *Open Id Secret* as in your OpenId provider
5. Set the *Open Id Code Challenge method* if you enabled authorization code encryption. In most cases, it is `S256`

> 6. Set *Open Id Login Claim*, *Open Id Email Claim*, *Open Id First Name Claim*, and *Open Id Last Name Claim* to provide optional claims for the application

Author: It should be more specific. Datagrok also requires email to be a part of claims.

Reviewer: @dnillovna  here it is 6.*Open Id Email Claim*  no?

> 7. You can enable/disable OpenID auto-login using the *Open Id Auto Login* checkbox

Author: Not worth mentioning, already clear from the UI

Reviewer: @dnillovna , All of this?

> 8. Make sure the correct **Web Root** is set in the left **Sidebar** > **Settings** > **Admin** section

Author: It is repeated in all instructions, lets move it to the top and make a general thing

## LDAP authentication

> You can integrate Datagrok with your LDAP server by enabling the smooth domain authentication mechanism across all your
> services. You can also use Active Directory to authenticate by this method.

Author: ```suggestion
To enable LDAP or Active Directory authentication:
```

1. On the left **Sidebar** go to **Settings** > **Users and Sessions**. Authentication settings page appears.

> 2. Enable *Domain authentication* to use the LDAP method

Author: > to use the LDAP method

It is already mentioned in the header, so can be omitted

> 3. Enable *Domain signup* to enable all users present on a domain controller to authenticate in the Datagrok platform.
>    If the option is disabled, it is required to create the user in the Datagrok platform first to allow the user to log into the platform

Author: ```suggestion
3. To enable all users on a domain controller to authenticate in the platform, use the *Domain signup* option. Otherwise, you need to create the user in the platform before they can log in.
```

> 4. Configure LDAP server address/DNS name

Author: Maybe more thorough instruction would be good. At lease for the last point about User.

> ## Oauth authentication

Author: the paragraph is outdated, and maybe should be split to 2/3 different paragraphs like in Settings.

Author: @ksumag did you consider what I suggested?
